src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,60 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURES_COUNT = 5;
+    private static final int MAX_NUMBER = 30;
+    private static final int DEFAULT_SIZE = 10;
+    private static final Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+
+        switch (random.nextInt(FIGURES_COUNT)) {
+            case 0:
+                return createRandomSquare();
+            case 1:
+                return createRandomCircle();
+            case 2:
+                return createRandomIsoscelesTrapezoid();
+            case 3:
+                return createRandomRectangle();
+            default:
+                return createRandomRightTriangle();
+        }
+    }
+
+    private Square createRandomSquare() {
+        return new Square(colorSupplier.getRandomColor(),
+        random.nextInt(MAX_NUMBER));
+    }
+
+    private Circle createRandomCircle() {
+        return new Circle(colorSupplier.getRandomColor(),
+        random.nextInt(MAX_NUMBER));
+    }
+
+    private RightTriangle createRandomRightTriangle() {
+        return new RightTriangle(colorSupplier.getRandomColor(),
+        random.nextInt(MAX_NUMBER),
+        random.nextInt(MAX_NUMBER));
+    }
+
+    private IsoscelesTrapezoid createRandomIsoscelesTrapezoid() {
+        return new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+        random.nextInt(MAX_NUMBER),
+        random.nextInt(MAX_NUMBER),
+        random.nextInt(MAX_NUMBER));
+    }
+
+    private Rectangle createRandomRectangle() {
+        return new Rectangle(colorSupplier.getRandomColor(),
+        random.nextInt(MAX_NUMBER),

```

Comment:
this is not property indented - use IDEA auto formatter, Ctrl Alt L for windows machines
also this can generate 0, and figure parameters can't be zero